bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue

```diff
@@ -328,156 +266,388 @@
+        custom="PROCEDURE" />
     </xml>
   </div>
 </template>
 
 <style lang="stylus">
-.blockly-editor
-  display block
-  top calc(100%)
-  height calc(100%)
-  width 100%
-  .blockly
-    height 100%
-    .blocklyMainBackground
-      stroke inherit
-.blocklyDropDownDiv
-  z-index 9000
+.blockly-editor {
+  display: block;
+  top: calc(100%);
+  height: calc(100%);
+  width: 100%;
+
+  .blockly {
+    height: 100%;
+
+    .blocklyMainBackground {
+      stroke: inherit;
+    }
+  }
+}
+
+.blocklyDropDownDiv {
+  z-index: 9000;
+}
 </style>
 
 <script>
 import Blockly from 'blockly'
+import '@blockly/block-plus-minus'
+
 import Vue from 'vue'
 
 import defineOHBlocks from '@/assets/definitions/blockly/ohblocks'
+import defineOHBlocksTimers from '@/assets/definitions/blockly/ohblocks_timers'
+import defineOHBlocksEphemeris from '@/assets/definitions/blockly/ohblocks_ephemeris'
+import defineOHBlocksAudio from '@/assets/definitions/blockly/ohblocks_audio'
+import defineOHBlocksBusEvents from '@/assets/definitions/blockly/ohblocks_busevents'
+import defineOHBlocksLogging from '@/assets/definitions/blockly/ohblocks_logging'
 
-Vue.config.ignoredElements = ['field', 'block', 'category', 'xml', 'mutation', 'value', 'sep']
+Vue.config.ignoredElements = [
+  'field',
+  'block',
+  'category',
+  'xml',
+  'mutation',
+  'value',
+  'sep'
+]
 
 export default {
   props: ['blocks'],
   data () {
     return {
       workspace: null,
-      loading: true
+      sinks: [],
+      voices: [],
+      scripts: [],
+      rules: [],
+      addons: [],
+      loading: true,
+      ready: false
     }
   },
-  mounted () {
-    defineOHBlocks(this.$f7)
-    this.workspace = Blockly.inject(this.$refs.blocklyEditor, {
-      toolbox: this.$refs.toolbox,
-      horizontalLayout: !this.$device.desktop,
-      theme: (this.$f7.data.themeOptions.dark === 'dark') ? 'dark' : undefined,
-      trashcan: false
-    })
-    const xml = Blockly.Xml.textToDom(this.blocks)
-    Blockly.Xml.domToWorkspace(xml, this.workspace)
-
-    this.workspace.addChangeListener(this.onChange)
-  },
-  beforeDestroy () {
-    this.workspace.removeChangeListener(this.onChange)
+  created () {
+    this.getAltData()
+    this.getAddons()
   },
   methods: {
+    startBlockly () {
+      this.workspace = Blockly.inject(this.$refs.blocklyEditor, {
+        toolbox: this.$refs.toolbox,
+        horizontalLayout: !this.$device.desktop,
+        theme: this.$f7.data.themeOptions.dark === 'dark' ? 'dark' : undefined,
+        trashcan: false
+      })
+      const xml = Blockly.Xml.textToDom(this.blocks)
+      Blockly.Xml.domToWorkspace(xml, this.workspace)
+    },
+    loadPage () {
+      defineOHBlocks(this.$f7)
+      defineOHBlocksTimers(this.$f7)
+      defineOHBlocksEphemeris(this.$f7)
+      defineOHBlocksLogging(this.$f7)
+      defineOHBlocksAudio(this.$f7, this.sinks, this.voices)
+      defineOHBlocksBusEvents(this.$f7)
+
+      this.startBlockly()
+    },
+    getAddons () {
```

Comment:
What is this for?

Reply:
As mentioned above this seems to be the only unused method but as I took over the code I am not really sure why it was put it. If I take it out we need to do a complete regression test.

Reply:
I would remove this, I don't think it's used anywhere.